### PR TITLE
JS API/Core/Data: Fix - commandExtractArgs.

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: core/common/assets/js/api/core/data.js [hunk 1]
```diff
@@ -478,9 +487,9 @@ export default class Data extends Commands {
 	run( type, command, args ) {
 		args.options.type = type;
 
-		command = this.commandExtractArgs( command, args );
+		const extractedCommand = this.commandExtractArgs( command );
```
Comment (matipojo) [inline core/common/assets/js/api/core/data.js:490]:
```suggestion
		const { command, args } = this.commandExtractArgs( command, args );
```

File: core/common/assets/js/api/core/data.js [hunk 2]
```diff
@@ -478,9 +487,9 @@ export default class Data extends Commands {
...
 
-		return super.run( command, args );
+		return super.run( extractedCommand.command, { ... args, ... extractedCommand.args } );
```
Comment (matipojo) [inline core/common/assets/js/api/core/data.js:492]:
```suggestion
		return super.run( command, args );
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — DISMISSED** [review]

File: core/common/assets/js/api/core/data.js [hunk 3]
```diff
@@ -189,27 +195,30 @@ export default class Data extends Commands {
...
 	 *
 	 * @param {string} command
-	 * @param {object} argsTarget
+	 * @param {object} args
 	 *
-	 * @returns {string} command
+	 * @returns {ExtractedCommand} command
 	 */
-	commandExtractArgs( command, argsTarget ) {
+	commandExtractArgs( command, args = {} ) {
```
Comment (matipojo) [inline core/common/assets/js/api/core/data.js:202]: Change also the PHP method.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/common/assets/js/api/core/data.js [hunk 4]
```diff
@@ -189,27 +195,30 @@ export default class Data extends Commands {
 	 * If the command have query convert it to args.
 	 *
 	 * @param {string} command
-	 * @param {object} argsTarget
+	 * @param {object} args
 	 *
-	 * @returns {string} command
+	 * @returns {ExtractedCommand} command
```
Comment (kobizz) [inline core/common/assets/js/api/core/data.js:200]: Is `ExtractedCommand` a type? It seems like plain object

File: core/common/assets/js/api/core/data.js (hunk 3 above)
Comment (kobizz) [inline core/common/assets/js/api/core/data.js:202]: Why do you need the args parameter at all? It's not related to this method purpose

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/common/assets/js/api/core/data.js (hunk 4 above)
Comment (iNewLegend) [reply core/common/assets/js/api/core/data.js:200]: Known object.

File: core/common/assets/js/api/core/data.js (hunk 3 above)
Comment (iNewLegend) [reply core/common/assets/js/api/core/data.js:202]: It used in cases, you want merge, args.

Comment (iNewLegend) [reply core/common/assets/js/api/core/data.js:202]: handled.

**Review by matipojo — APPROVED** [review]

File: core/common/assets/js/api/core/data.js (hunk 4 above)
Comment (kobizz) [reply core/common/assets/js/api/core/data.js:200]: As far as I know, only instances object should use this kind of typing. Could you provide me with a reference?

Comment (matipojo) [reply core/common/assets/js/api/core/data.js:200]: @kobizz see the typedef above:
https://github.com/elementor/elementor/pull/11660/files#diff-d0a28344df7ce0852c5a7c797db34c66R21

Comment (iNewLegend) [reply core/common/assets/js/api/core/data.js:200]: this is how jsdoc works its create 'fake' type.

**Review by kobizz — APPROVED** [review]

**Review by KingYes — APPROVED** [review]

**Pull request merged**